vm/devices/storage/storage_tests/tests/scsidvd_nvme.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+//! Tests using NVMe as the block backend for SimpleScsiDvd.
+#![cfg(target_os = "linux")]
+
+use chipset_device::mmio::ExternallyManagedMmioIntercepts;
+use disk_backend::Disk;
+use disk_nvme::NvmeDisk;
+use guestmem::GuestMemory;
+use guid::Guid;
+use nvme::NvmeController;
+use nvme::NvmeControllerCaps;
+use nvme_driver::NvmeDriver;
+use pal_async::DefaultDriver;
+use pal_async::async_test;
+use pci_core::msi::MsiInterruptSet;
+use scsi_buffers::OwnedRequestBuffers;
+use scsi_buffers::RequestBuffers;
+use scsi_core::AsyncScsiDisk;
+use scsi_core::Request;
+use scsi_defs::ISO_SECTOR_SIZE;
+use scsi_defs::ScsiOp;
+use scsidisk::scsidvd::SimpleScsiDvd;
+use test_with_tracing::test;
+use user_driver::emulated::DeviceSharedMemory;
+use user_driver::emulated::EmulatedDevice;
+use vmcore::vm_task::SingleDriverBackend;
+use vmcore::vm_task::VmTaskDriverSource;
+use zerocopy::IntoBytes;
+
+struct ScsiDvdNvmeTest {
+    scsi_dvd: SimpleScsiDvd,
+    _nvme_driver: NvmeDriver<EmulatedDevice<NvmeController>>, // We need to store this to keep it from going out of scope
+}
+
+impl ScsiDvdNvmeTest {
+    async fn new(
+        driver: DefaultDriver,
+        sector_size: u32,
```

Comment:
What's a bummer is that we don't actually create a device with this sector size. The `ram_disk` hard codes 512B sectors. Please take a follow-up work item to add 4K support here.

Reply:
I will look at this in a follow up task 😃

Reply:
Created #1492